dfastmi/batch/AreaPlotter.py

```diff
@@ -0,0 +1,388 @@
+import dfastmi.plotting
+from dfastmi.batch.AreaDetector import AreaDetector
+from dfastmi.batch.PlotOptions import PlotOptions
+
+
+class AreaPlotter(ABC):
+    """
+    Abstract class used to plot the area.
+    """
+
+    _total_str: str
+    _area_str: str
+    _positive_up: bool
+
+    def __init__(
+        self,
+        plotting_options: PlotOptions,
+        plot_n: int,
+        area_detector: AreaDetector,
+        xyz_file_location: Path,
+    ):
+        """
+        Arguments
+        ---------
+        plotting_options : PlotOptions
+            Options for plotting.
+        area_detector: AreaDetector
+            Class which holds the information regarding detected areas.
+        plot_n : int
+            n for plotting.
+        xyz_file_location : Path
+            Location to write the xyz file to.
+        """
+        self._xyz_file_location = xyz_file_location
+        self._plotting_options = plotting_options
+        self._plot_n = plot_n
+        self._area_detector = area_detector
+
+    def plot_areas(
+        self,
+        dzgemi: numpy.ndarray,
+        areai: numpy.ndarray,
+        wbin: numpy.ndarray,
+        wbin_labels: list[str],
+        wthresh: numpy.ndarray,
+        siface: numpy.ndarray,
+        afrac: numpy.ndarray,
+        sbin: numpy.ndarray,
+        sthresh: numpy.ndarray,
+        kmid: numpy.ndarray,
+    ):
+        """
+        Plot the area data.
+
+        Arguments
+        ---------
+        dzgemi : numpy.ndarray
+            Yearly mean bed level change [m].
+        areai : numpy.ndarray
+            Array of length M containing the grid cell area [m2].
+        wbin: numpy.ndarray
+            Array of length N containing the index of the target width bin [-].
+        wbin_labels: list[str]
+            Array of length N containing the index of the target width bin [-].
+        wthresh : numpy.ndarray
+            Array containing the cross-stream coordinate boundaries between the width bins [m].
+        siface : numpy.ndarray
+            Array of length N containing the index of the source cell (range 0 to M-1) [-].
+        afrac : numpy.ndarray
+            Array of length N containing the fraction of the source cell associated with the target chainage bin [-].
+        sbin : numpy.ndarray
+            Array of length N containing the index of the target chainage bin [-].
+        sthresh : numpy.ndarray
+            Array containing the along-stream coordinate boundaries between the streamwise bins [m].
+        kmid: numpy.ndarray
+            Array of length N containing the location of points expressed as chainage.
+        """
+
+        sbin_length = sthresh[1] - sthresh[0]
+        binvol = self._comp_binned_volumes(
+            numpy.maximum(dzgemi, 0.0),
+            areai,
+            wbin,
+            siface,
+            afrac,
+            sbin,
+            wthresh,
+            sthresh,
+        )
+
+        self._write_xyz_file(wbin_labels, kmid, binvol)
```

Comment:
This is a sneaky little side-effect of the `plot_areas` function. Maybe we can check if we can move this out of this function and place at a higher level.

Reply:
agreed